### ISSUE TITLE
report race condition bug upstream

### ISSUE THREAD

**Issue by adrelanos** [grml/grml-debootstrap#36]
From
https://github.com/grml/grml-debootstrap/blob/master/grml-debootstrap#L938

```
# give the system 2 seconds, otherwise we might run into
# race conditions :-/
sleep 2
```

Is there an upstream bug report about this? Could you report that one please?

I tried reporting an eventually similar/related bug a while ago:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=734794

Without much success.

In meanwhile things got worse. Now I need to `sleep 2` before and after using kpartx.

Perhaps you have better chances of getting through?

**Comment by mika** [comment]
Patrick, did you notice any further things with this issue? I'd love to get rid of this 'sleep 2' workaround, but am unsure whether that's "safe" (as dirty as it is :-/).

**Comment by adrelanos** [comment]
Unfortunately not. I don't think you remove the 'sleep 2' workaround until https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=734794 has been sorted out.

This ticket #36 was my call for help getting this reported upstream so that workaround can one day be removed from grml-deboostrap. Not sure my above bug report https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=734794 is good enough and will lead one day to something.

Basically I was wondering if you could come up with instructions for reproduction of this issue, that upstream and/or Debian would use, so they'll get aware of the issue.

**Comment by mika** [comment]
Ok thanks, I need to find some spare time to work on this, as long as it's working for grml-debootstrap priority is quite low though, jftr.

**Comment by mika** [comment]
Seems to be taken care of in upstream code, please let us know if it's still an issue and we need to do something about it. Thanks.
